### PR TITLE
Release snapshot versions automatically using Jenkins

### PR DESCRIPTION
**Pull request by matir91**

### CODE REVIEW

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+***Pending Image***
+
+Now all "Build" actions were made, so now we have to add a new "Post-Build Action" in order to release the build using Beta. For that we will use 2 plugins,  [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin) to check that the build branch is `develop` and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
+
+<!-- Image with the code -->
+***Pending Image***
+
+Using this configuration, the `SNAPSHOT` build will be upload, it will have a useful release notes and all your team will be notified. 
+
+So, making that, you will be releasing each new feature ASAP without effort and somebody could test it and maybe find a bug could be much easier.
```
Comment (m-revetria) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `So, by doing this, ...` instead of `So, making that, ...` ??

Comment (m-revetria) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `maybe find ...` ~> `maybe finding ...` or `maybe to find ...` ??

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+In order to release a new `SNAPSHOT` version, first of all we should make some work:
+* Check that the build branch is `develop`
+* Update the build `versionName` in order to track possible issues with the right version, in this example the version name will be built using the app current versionName, plus the current hash of the commit.
+* Create a release note file, including for example:
+  * Commit message
+  * Author name  
+  * Last 10 commits (in order to see what are the last changes introduced)
+  * And all things that you want
+
+In order to make this work we should create a new build "Execute shell" build task before the compilation, run tests, check lints and that stuff.
```
Comment (matir91) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: hahah my bad I wanted to delete it, but I forgot it.

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+This script is public in [github](https://gist.github.com/matir91/5a8c24196c0fd4408adaabfdab6f198a)
+
+After that we should add the `Invoke gradle script` task, which should have all gradle task that we usually do (compile, run tests, check lints, etc.). Additional we should add a new gradle task, in order to make the release apk. However, we should be careful, it should be done only if the current build is from the `develop` branch. So we should define another build task to compile the new release build, if the new commit comes from the `develop` branch.
+
+First of all, we should inject the properties file that we made, in the system environment variables, for that, we will use the  [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin).
+ 
+<!-- Image with the code -->
+***Pending Image***
+
+Then we should add a condition build task, in order to generate the release apk only if build branch is `develop`, for doing that, we will use the [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin).
```
Comment (matir91) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: The plugin is `conditional`, so I think I should remplace the `condition` to `conditional`

Comment (michaelEllisUy) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: IMO `conditional` is 👍

Comment (michaelEllisUy) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `is `develop`, for doing that, we will use the` = is `develop`. For this, we will use...

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+<!-- Image with the code -->
+***Pending Image***
+
+Now all "Build" actions were made, so now we have to add a new "Post-Build Action" in order to release the build using Beta. For that we will use 2 plugins,  [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin) to check that the build branch is `develop` and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
```
Comment (michaelEllisUy) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `Now all "Build" actions were made, so now we have to add a new "Post-Build Action"` Now that all "Build" actions were done, we have to add a new...

Comment (michaelEllisUy) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `For that we will use 2 plugins,` In this case, we'll use 2 plugins

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+<!-- Image with the code -->
+***Pending Image***
+
+Using this configuration, the `SNAPSHOT` build will be upload, it will have a useful release notes and all your team will be notified.
```
Comment (michaelEllisUy) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `upload, it will have a useful release notes and all your team will be notified. ` = uploaded, will have useful release notes and all your team will be notified

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 1 above)
Comment (michaelEllisUy) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `without effort` = with NO effort at all, and anyone could test it to find bugs at an earlier stage, as well as making it easier to find them.

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+layout: post
+title:  Release snapshot versions automatically in Jenkins
+date:   2017-01-30 10:23:24
+author: Matías Irland
+categories: Android,Server,CI,Jenkins
+author_id: mirland
+
+---
+
+Everyone knows that if you are a developer, a CI server could "change your life". Currently in the Xmartlabs Android team, we are using jenkins, which is an amazing tool, and is helping us in all our build tasks.
```
Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: does everybody know? I think that's not the case. Don't know if explaining that is out of scope

Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `Jenkins`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+Today I want to tell you how we can release `SNAPSHOT` versions using [Beta of Fabric](https://docs.fabric.io/android/beta/overview.html).
```
Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `Beta by Crashlytics`

Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Maybe a brief explanation of what it is? I'm just thinking a one liner

Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: I think we should explain here the benefits of doing that

Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: And maybe a very brief explanation of what a snapshot is. I think at this very point in the blog a user skimming it will decide if it's worth reading, so a good paragraph here is key

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+author: Matías Irland
+categories: Android,Server,CI,Jenkins
+author_id: mirland
+
+---
+
+Everyone knows that if you are a developer, a CI server could "change your life". Currently in the Xmartlabs Android team, we are using jenkins, which is an amazing tool, and is helping us in all our build tasks.
+
+Today I want to tell you how we can release `SNAPSHOT` versions using [Beta of Fabric](https://docs.fabric.io/android/beta/overview.html). 
+Suppose that in your git repository, you have a `develop` branch, which is the *main branch* of this repository, which has all changes ready to release. However, most of us usually group a bunch of features or wait to introduce a "big" change before releasing. The main thing here, is that we have changes ready to release, which could be tested for you or someone in your team, but are not released, mainly because of the required effort and time consumption. If it's your situation, this post could be useful for you.
```
Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: To `which` in a row kinda slows the reading. What about:
`Consider your repository main development branch, which contains all the changes ready for a release`?

Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Well, now that i'm reading it, the first line sounds like you have a bunch of changes to release, which is not the point of releasing a snapshot. What about tackling it from the perspective of: `Suppose you just merged a feature into your main development branch...`

Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `If this is your situation...`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+First of all, lets check which [jenkins plugins](https://wiki.jenkins.io/display/JENKINS/Plugins) will be required for this work:
+1. [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin)
+2. [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin)
+3. [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin)
+
+
+Well, now we're ready to go.
+
+Firstly, in order to release a new `SNAPSHOT`, we should do some work:
```
Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: `we need to do some work`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+In order to make this work, we should create a new "Execute shell" build task before the compilation, run tests, check lints and that stuff.
+
+<!-- Image with the code -->
+***Pending Image***
+
+This script is public in [github](https://gist.github.com/matir91/5a8c24196c0fd4408adaabfdab6f198a)
+
+Afterwards, we have to add the `Invoke gradle script` build task, which should have all gradle tasks that we usually do (compile, run tests, check lints, etc.). In addition, we should add a new gradle task, in order to make the release apk. However, we should be careful, it should only be done if the current build is from the `develop` branch. In consequence, we have to define another build task to compile the new release build if the new commit comes from the `develop` branch.
+
+First of all, we should inject the properties file that we created in the system environment variables. To do that, we will use the  [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin).
```
Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: What about using subtitles to break this second `First, Firstly...`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+***Pending Image***
+
+Now that all "Build" actions were done, we have to add a new "Post-Build Action" in order to release the build using Beta. In this case, we'll use 2 plugins,  [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin) to check that the build branch is `develop` and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
+
+<!-- Image with the code -->
+***Pending Image***
+
+Using this configuration, the `SNAPSHOT` build will be uploaded, will have useful release notes and all your team will be notified. 
+
+So, by doing this, you will be releasing each new feature ASAP with NO effort at all, and anyone could test it to find bugs at an earlier stage, as well as making it easier to find them.
```
Comment (diegomedina248) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: remove the `So`, it'll look cleaner

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 7 above)
Comment (matir91) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: crashlitycs is a service of fabric, beta is another service of fabric. I think we should keep Fabric. What  do you think @diegomedina248 ?

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 8 above)
Comment (matir91) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: In the first comment I think we should keep the branch name, is easier to reference that in the rest of the post, How could we do?

**Comment by bryant1410** [comment]
Add yourself as an author in `_data/authors.yml`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+layout: post
+title:  Release snapshot versions automatically in Jenkins
+date:   2017-01-30 10:23:24
+author: Matías Irland
+categories: Android,Server,CI,Jenkins
+author_id: mirland
+
+---
+
+Today I want to show how we can release `SNAPSHOT` versions using [Beta by Fabric](https://docs.fabric.io/android/beta/overview.html).
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: I'd change these backticks for asterisks. The backticks are for code-related stuff in general. These are just names.

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title:  Release snapshot versions automatically in Jenkins
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: What about "Easy Continuous Delivery in Android with Jenkins"?

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+date:   2017-01-30 10:23:24
+author: Matías Irland
+categories: Android,Server,CI,Jenkins
+author_id: mirland
+
+---
+
+Today I want to show how we can release `SNAPSHOT` versions using [Beta by Fabric](https://docs.fabric.io/android/beta/overview.html).
+Have you ever been in a situation where you release new features between long periods of time and find yourself with tons of bugs to deal with? This process could be a lot smoother if you consider each change you merge into your *main development branch* to be ready for release. Once you're at that stage, you could automatically and effortlessly generate and release a new build of your application for each merge, that could be tested earlier. Moreover, you could run tests, check lints and any other task you normally do, automatically (Yes, magic, i know ;)). That's the beauty of having a CI server. It could "change your life".
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: This process -> The process ?

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Why stressing "main development branch"?

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Yes -> yes

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: i know -> I know

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: ;)) -> ;) (even if it's a closing parenthesis, I think it looks better)

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: CI server -> CI/CD server

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: It could "change your life" -> Saves the day

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+
+We're going to use `Jenkins`, an amazing CI server, to help us release each one of those changes. Each release generates what is called a `SNAPSHOT`. Our goal is to release something that is useful, not just the build. For that reason, we'll need the following `Jenkins` plugins to get started:
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: CI/CD

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+Have you ever been in a situation where you release new features between long periods of time and find yourself with tons of bugs to deal with? This process could be a lot smoother if you consider each change you merge into your *main development branch* to be ready for release. Once you're at that stage, you could automatically and effortlessly generate and release a new build of your application for each merge, that could be tested earlier. Moreover, you could run tests, check lints and any other task you normally do, automatically (Yes, magic, i know ;)). That's the beauty of having a CI server. It could "change your life".
+
+We're going to use `Jenkins`, an amazing CI server, to help us release each one of those changes. Each release generates what is called a `SNAPSHOT`. Our goal is to release something that is useful, not just the build. For that reason, we'll need the following `Jenkins` plugins to get started: 
+
+First of all, lets check which [jenkins plugins](https://wiki.jenkins.io/display/JENKINS/Plugins) will be required for this work:
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Jenkins

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: this work -> the work

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: lets -> let's

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: check -> see

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Leave a new line, as it looks broken in the post

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+1. [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin)
+2. [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin)
+3. [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin)
+
+
+Well, now we're ready to go.
+
+Firstly, in order to release a new `SNAPSHOT`, we need to do some work:
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Same paragraph as the one above?

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Leave a newline for it to work

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+* Check that the build branch is `develop`
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: develop -> develop (or the one you use for main development)

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+* Update the build `versionName` in order to track possible issues with the right version. In this example the version name will be built using the app current `versionName`, plus the current hash of the commit.
+* Create a release note file, including for example:
+  * Commit message
+  * Author name  
+  * Last 10 commits (in order to see what are the last changes introduced)
+  * And all things that you want
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Remove this line

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+
+In order to make this work, we should create a new "Execute shell" build task before the compilation, run tests, check lints and that stuff.
+
+<!-- Image with the code -->
+***Pending Image***
+
+This script is public in [github](https://gist.github.com/matir91/5a8c24196c0fd4408adaabfdab6f198a)
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: in github -> on GitHub

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: what about linking from "script" to "github" instead of just "github"?

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 12 above)
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: clarify that this is for Android

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+***Pending Image***
+
+Now that all "Build" actions were done, we have to add a new "Post-Build Action" in order to release the build using Beta. In this case, we'll use 2 plugins,  [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin) to check that the build branch is `develop` and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
+
+<!-- Image with the code -->
+***Pending Image***
+
+Using this configuration, the `SNAPSHOT` build will be uploaded, will have useful release notes and all your team will be notified. 
+
+By doing this, you will be releasing each new feature ASAP with NO effort at all, and anyone could test it to find bugs at an earlier stage, as well as making it easier to find them.
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: ASAP -> asap

Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: NO -> *no*

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 19 above)
Comment (matir91) [reply _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: why? maybe you want to add something more, for example a custom text or something like that

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 14 above)
Comment (matir91) [reply _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: don't you prefer `It could "change your life"` ? I don't like too much " Saves the day", something more?

Comment (bryant1410) [reply _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Don't know. Ask @joaquinrok

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+author: Matías Irland
+categories: Android,Server,CI,Jenkins,Fabric
+author_id: mirland
+
+---
+
+Today I want to show how we can release Android *SNAPSHOT* versions using [Beta by Fabric](https://docs.fabric.io/android/beta/overview.html).
+Have you ever been in a situation where you release new features between long periods of time and find yourself with tons of bugs to deal with? The process could be a lot smoother if you consider each change you merge into your main development branch to be ready for release. Once you're at that stage, you could automatically and effortlessly generate and release a new build of your application for each merge, that could be tested earlier. Moreover, you could run tests, check lints and any other task you normally do, automatically (yes, magic, I know ;). That's the beauty of having a CI/CD server. It could "change your life".
+
+We're going to use `Jenkins`, an amazing CI/CD server, to help us release each one of those changes. Each release generates what is called a `SNAPSHOT`. Our goal is to release something that is useful, not just the build. For that reason, we'll need the following `Jenkins` plugins to get started:
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: still with backticks?

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+title:  Easy Continuous Delivery in Android with Jenkins
+date:   2017-01-30 10:23:24
+author: Matías Irland
+categories: Android,Server,CI,Jenkins,Fabric
+author_id: mirland
+
+---
+
+Today I want to show how we can release Android *SNAPSHOT* versions using [Beta by Fabric](https://docs.fabric.io/android/beta/overview.html).
+Have you ever been in a situation where you release new features between long periods of time and find yourself with tons of bugs to deal with? The process could be a lot smoother if you consider each change you merge into your main development branch to be ready for release. Once you're at that stage, you could automatically and effortlessly generate and release a new build of your application for each merge, that could be tested earlier. Moreover, you could run tests, check lints and any other task you normally do, automatically (yes, magic, I know ;). That's the beauty of having a CI/CD server. It could "change your life".
```
Comment (patriciapoj2017) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Faltaría un ) para cerrar el paréntesis -la carita igual se ve como emoji en el blog, no?)

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+
+First of all, we should inject the properties file that we created in the system environment variables. To do that, we will use the  [EnvInject Plugin](https://wiki.jenkins.io/display/JENKINS/EnvInject+Plugin).
+ 
+<img src="/images/jenkins-snapshot/2-inject_variables.png" />
+
+Then we should add a conditional build task, in order to generate the release apk only if build branch is _develop_. For this, we will use the [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin).
+
+<img src="/images/jenkins-snapshot/3-build.png" />
+
+Now that all "Build" actions were done, we have to add a new "Post-Build Action" in order to release the build using Beta. In this case, we'll use 2 plugins,  [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin) to check that the build branch is _develop_ and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
```
Comment (patriciapoj2017) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: Yo pondría build actions have been done

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+
+<img src="/images/jenkins-snapshot/4-release.png" />
+
+Using this configuration, the _SNAPSHOT_ build will be uploaded, will have useful release notes and all your team will be notified. 
+
+By doing this, you will be releasing each new feature asap with *no* effort at all, and anyone could test it to find bugs at an earlier stage, as well as making it easier to find them.
```
Comment (patriciapoj2017) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: yo pondría anyone will be able to test it, ya que estamos hablando en futuro

Comment (matir91) [reply _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: shouldn't we add a comma too? `anyone will be able to test it*,* to find bugs at an earlier stage`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md (hunk 23 above)
Comment (matir91) [reply _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: nop, it will be displayed as a text, that is the reason why we used only one `)`. Do yo you prefer `(yes, magic, I know ;)` instead of `(yes, magic, I know ;))`

File: _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md [hunk 26]
```diff
@@ -48,7 +49,7 @@ Then we should add a conditional build task, in order to generate the release ap
 
 <img src="/images/jenkins-snapshot/3-build.png" />
 
-Now that all "Build" actions have been done, we have to add a new "Post-Build Action" in order to release the build using Beta. In this case, we'll use 2 plugins,  [Conditional BuildStep Plugin](https://wiki.jenkins.io/display/JENKINS/Conditional+BuildStep+Plugin) to check that the build branch is _develop_ and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
+Now that all "Build" actions have been done, we have to add a new "Post-Build Action" in order to release the build using Beta. In this case, we'll use 2 plugins,  [Flexible Publish Plugin](https://wiki.jenkins.io/display/JENKINS/Flexible+Publish+Plugin) to check that the build branch is _develop_ and [Fabric Beta Publisher Plugin](https://wiki.jenkins.io/display/JENKINS/Fabric+Beta+Publisher+Plugin) to upload the build. Furthermore, we will save the generated apk in the release build information using the "Archive the artifacts" post-build task.
```
Comment (bryant1410) [inline _posts/2017-06-27-Release-Snapshot-Versions-In-Jenkins.md]: double space
